tests/mock_crust/churn.rs

```diff
@@ -213,6 +253,7 @@ impl ExpectedGets {
                 }
             }
         }
+        assert!(unexpected_receive <= self.sections.len());
```

Comment:
Since it's at most one for each message, let's not just count, but instead record the messages itself and assert that each of them was received in at most one unexpected node.